### PR TITLE
Allow clients to report if they own or share the underlying connection string (Part 2)

### LINKED ISSUE #6032: [Service Bus] [Transferred issue] Allow clients to report if they own or share the underlying connection string
@SeanFeldman  please add details.

### PR DESCRIPTION
**Pull request by SeanFeldman**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/6032

Breaking change. Should be merged for the next **major**.

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@jsquire @AlexGhiondea have you had a chance to triage this one?

**Comment by SeanFeldman** [comment]
Ping

**Comment by jsquire** [comment]
I don't have good insight into the current plans for the next Service Bus release, as I'm focused elsewhere at present.   I'll need to defer to @AlexGhiondea, who will have better insight.

**Comment by AlexGhiondea** [comment]
Hey @SeanFeldman! As we don't yet have sufficient context on this, I am adding @binzywu and @makam to take a look.

**Comment by SeanFeldman** [comment]
@AlexGhiondea have you had a chance to connect with the broker team?
Would be nice to have this PR labelled with `Client`, `Service Bus`, and `breaking-change` to indicate what it stands for.

**Comment by jsquire** [comment]
I'll have to continue to defer to @AlexGhiondea for insight regarding conversations with the service team.   However, I assigned the issue, applied tags, and tagged reviewers.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/IClientEntity.cs [hunk 1]
```diff
@@ -44,6 +44,11 @@ public interface IClientEntity
         /// </summary>
         ServiceBusConnection ServiceBusConnection { get; }
 
+        /// <summary>
+        /// Returns true if connection is owned and false if connection is shared.
+        /// </summary>
+        bool OwnsConnection { get; }
```
Comment (jsquire) [inline sdk/servicebus/Microsoft.Azure.ServiceBus/src/IClientEntity.cs]: For context, this property exists on the `ClientEntity` base class, but was not exposed via the interface.  Unless I overlooked something, this does not appear to require any further implementation on types exposing the interface.

The breaking change is to the interface contract only.

**Review by jsquire — APPROVED** [review]
Assuming that my comment on the `IClientEntity` type is correct, this looks like a good idea to me, for the next major.

@makam:  I'd appreciate your thoughts.

File: sdk/servicebus/Microsoft.Azure.ServiceBus/src/IClientEntity.cs (hunk 1 above)
Comment (SeanFeldman) [reply sdk/servicebus/Microsoft.Azure.ServiceBus/src/IClientEntity.cs]: That's correct. `OwnsConnection` was added to all `IClientEntity` implementors and not the interface itself to allow the fix in a minor release. This issue will complete the implementation by extracting it into the contract, while still having the actual implementation on place. Hence the breaking change.

**Comment by jsquire** [comment]
@SeanFeldman:  I didn't see any objections and the change seems reasonable to me;  I merged it in so that it'll go with the pending set of breaking changes.

**Pull request merged**